src/accountsdb/db.zig

```diff
@@ -103,6 +103,11 @@ pub const AccountsDB = struct {
     latest_full_snapshot_info: RwMux(?FullSnapshotGenerationInfo) = RwMux(?FullSnapshotGenerationInfo).init(null),
     latest_incremental_snapshot_info: RwMux(?IncSnapshotGenerationInfo) = RwMux(?IncSnapshotGenerationInfo).init(null),
 
+    account_hashes_and_lamports_cache: struct {
+        first_full: RwMux(?FullSnapshotGenerationInfo) = RwMux(?FullSnapshotGenerationInfo).init(null),
```

Comment:
Well, they don't act as the latest, they only act as the first. They are initialized after validating a snapshot load, so they're always only the first, which is the only time they may be useful.

Reply:
what if i loaded from a snapshot twice

Reply:
When would you want to validate a snapshot twice?

Reply:
not sure - but its possible - and latest is accurate in both scenarios so is a better name imo

Reply:
But functionally it would still be the first in the success path: if you validate against a snapshot more than once, and the second time you're validating against a *different* snapshot than the one you actually loaded from, it would fail. If you validate against the same one, it would succeed, and no information actually becomes new, it would remain the same as it was.

The only scenario I could imagine where this would even be marginally possible is if you wanted to validate against a snapshot later on after having processed updates to accounts, to see if accountsdb is in sync with that new snapshot - but we don't actually support this because the function assumes we *just* loaded from the snapshot (ie we don't specify the concrete range for the computation, just the min xor max slot).